test/interceptors/request.test.ts

```diff
@@ -149,11 +150,8 @@ describe('test request interceptor', () => {
     expect(response2.status).toBe(200);
   });
 
-  it('tests last modified header handling with modified-since date', async () => {
-    const axios = mockAxios(
-      {},
-      { 'last-modified': 'Wed, 21 Oct 2015 07:28:00 GMT', 'cache-control': 'max-age=1' }
-    );
+  it('tests last modified header handling with modifiedSince date', async () => {
```

Comment:
all of these tests makes no sense to me. 

Why would you want a manual date when there's one in the header, why is there no test with no date in header ?

I'll try to make these manual values work by overriding the response headers so they end up in cache. However, this is kinda ugly. Do you have a use case in mind or are you simply trying to always have the ability to provide custom values ? These two (etag and last-modified) are only used for revalidation and not for caching, so it would be best to not allow to customize them, and leave values customization to what used for caching (namely cache time (ttl) which you have already and grace time (name used in varnish-cache) when you use stale value during revalidation (feature to come later)

Reply:
Thinking on this, I can probably add 2 dedicated headers for this, like `x-axios-cache-last-modified` and `x-axios-cache-etag` and use those alongside the real headers